acceptance/tests/resource/cron/should_only_fail_associated_resources_on_file_read_error.rb

```diff
@@ -0,0 +1,117 @@
+      stub_crontab_exe = "#{stub_crontab_bin_dir}/crontab"
+
+      # On Linux and OSX, we read a user's crontab by running crontab -u <username>,
+      # where the crontab command is run as root. However on AIX/Solaris, we read a
+      # user's crontab by running the crontab command as that user. Thus our mock
+      # crontab executable needs to check if we're reading our failed user's crontab
+      # (Linux and OSX) OR running crontab as our failed user (AIX and Solaris) before
+      # triggering the FileReadError
+      stub_crontab_exe_script = <<-SCRIPT
+#!/usr/bin/env bash
+
+if [[ "$@" =~ #{failed_username} || "`id`" =~ #{failed_username} ]]; then
+  echo "Mocking a FileReadError for the #{failed_username} user's crontab!"
+  exit 1
+fi
+
+#{crontab_exe} $@
+SCRIPT
+
+      create_remote_file(agent, stub_crontab_exe, stub_crontab_exe_script)
+
+      on(agent, "chmod 777 #{stub_crontab_bin_dir}")
+      on(agent, "chmod 777 #{stub_crontab_exe}")
+    end
+
+    path_env_var = nil
+    step "Get the value of the PATH environment variable" do
+      path_env_var = on(agent, "echo $PATH").stdout.chomp
+    end
+
+    puppet_result = nil
+    step "Add some cron entries with Puppet" do
+      # We delete our mock crontab executable here to ensure that Cron[second_entry]'s
+      # evaluation fails because of the FileReadError raised in the prefetch
+      # step. Otherwise, Cron[second_entry]'s evaluation will fail at the write step
+      # because Puppet would still be invoking our mock crontab executable, which would
+      # pass the test on an agent that swallows FileReadErrors in the cron provider's
+      # prefetch step.
+      manifest = [
+        cron_manifest('first_entry', command: "ls", user: username),
+        file_manifest(stub_crontab_exe, ensure: :absent),
+        cron_manifest('second_entry', command: "ls", user: failed_username),
+      ].join("\n\n")
+      manifest_file = agent.tmpfile("crontab_overwrite_manifest")
+      create_remote_file(agent, manifest_file, manifest)
+
+      # We need to run a script here instead of a command because:
+      #   * We need to cd into a directory that our user can access. Otherwise, bash will
+      #   fail to execute stub_crontab_exe on AIX and Solaris because we run crontab
+      #   as the given user, and the given user does not have access to Puppet's cwd.
+      #
+      #   * We also need to pass-in our PATH to Puppet since it contains stub_crontab_bin_dir.
+      apply_crontab_overwrite_manifest = agent.tmpfile("apply_crontab_overwrite_manifest")
+      script = <<-SCRIPT
+#!/usr/bin/env bash
+
+cd #{stub_crontab_bin_dir} && puppet apply #{manifest_file} 
+SCRIPT
+      create_remote_file(agent, apply_crontab_overwrite_manifest, script)
+      on(agent, "chmod a+x #{apply_crontab_overwrite_manifest}")
+
+      puppet_result = on(agent, "bash #{apply_crontab_overwrite_manifest}", environment: { PATH: "#{stub_crontab_bin_dir}:#{path_env_var}" })
```

Comment:
We do need to use a script, because the PATH variable does not persist in the `&&`. Specifically, what happens is:
```
env PATH="/tmp/stub_crontab_bin_dir.tEaOlQ:$PATH:/opt/csw/bin:/usr/sbin:/usr/bin:/opt/puppetlabs/bin" cd /tmp/stub_crontab_bin_dir.tEaOlQ && puppet apply /tmp/crontab_overwrite_manifest.r8aqmQ
```

is what's run, but only the `cd` gets the PATH variable. Our puppet apply command doesn't.